checker/src/org/checkerframework/checker/lock/LockStore.java

```diff
@@ -137,4 +159,83 @@ protected void internalDotOutput(StringBuilder result) {
                 + "\\n");
         super.internalDotOutput(result);
     }
+
+    @Override
+    protected boolean isSideEffectFree(AnnotatedTypeFactory atypeFactory,
+            ExecutableElement method) {
+        return ((LockAnnotatedTypeFactory) atypeFactory).methodSideEffectAnnotation(method, false) == SideEffectAnnotation.RELEASESNOLOCKS ||
+               super.isSideEffectFree(atypeFactory, method);
+    }
+
+    // Same as super method, except for the call to localVariableValues.clear();
+    // See the comment next to the call for more details.
+    @Override
+    public void updateForMethodCall(MethodInvocationNode n,
+            AnnotatedTypeFactory atypeFactory, CFValue val) {
+        ExecutableElement method = n.getTarget().getMethod();
+
+        // case 1: remove information if necessary
+        if (!(((LockChecker) analysis.getTypeFactory().getContext()).hasOption("assumeSideEffectFree")
```

Comment:
Copying a method's body and making a small change is not an ideal way of overriding it. In this particular method (updateForMethodCall), the only change I needed to make was the call to localVariableValues.clear() - that is the only reason I am overriding that method here. Keeping the changes to updateForMethodCall as small as possible will make it easier to remove duplicate code later.